components/brave_shields/core/browser/ad_block_component_installer.cc

```diff
@@ -143,56 +148,59 @@ void OnRegistered(const std::string& component_id) {
       component_id, component_updater::OnDemandUpdater::Priority::FOREGROUND);
 }
 
-}  // namespace
-
-void RegisterAdBlockDefaultResourceComponent(
+void RegisterAdBlockComponentInternal(
     component_updater::ComponentUpdateService* cus,
-    OnComponentReadyCallback callback) {
+    const std::string& component_public_key,
+    const std::string& component_id,
+    const std::string& component_name,
+    OnSecureComponentReadyCallback callback) {
   // In test, |cus| could be nullptr.
   if (!cus) {
     return;
   }
 
+  auto on_ready = base::BindRepeating(
+      [](OnSecureComponentReadyCallback callback, const base::FilePath& path) {
+        brave_component_updater::ComponentContentsVerifier::GetInstance()
```

Comment:
We register each component only once, but it is updated **multiple times** during the browser lifetime.
I's not absolutely clear whether we reuse the same `ComponentContentsVerifier` or not in this case.
Looks like we load `verified_contents_` in the ctor and this callback is called once after the component is installed/loaded.

Also it makes sense to explicitly clarify it in the class description.

Reply:
We do not reuse the accessor, for different components paths it recreated